main.go

```diff
@@ -82,9 +107,70 @@ func (c *Cmd) Run() {
 	if err := c.cmd.Run(); err != nil {
 		c.exitError = err
 	}
-	c.stdout = string(outBuf.Bytes())
-	c.stderr = string(errBuf.Bytes())
-	c.executed = true
+	c.stdout.content = string(outBuf.Bytes())
+	c.stderr.content = string(errBuf.Bytes())
+	c.status = "executed"
+}
+
+// Start starts the command without waiting for it to complete
+func (c *Cmd) Start() {
+	if c.stdin != nil {
+		c.cmd.Stdin = c.stdin
+	}
+
+	if c.env != nil {
+		c.cmd.Env = c.env
+	} else {
+		c.cmd.Env = os.Environ()
+	}
+
+	stdoutPipe, err := c.cmd.StdoutPipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	stderrPipe, err := c.cmd.StderrPipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := c.cmd.Start(); err != nil {
+		c.exitError = err
+	}
+
+	go func() {
+		scanner := bufio.NewScanner(stdoutPipe)
+		for scanner.Scan() {
+			c.stdout.mu.Lock()
+			c.stdout.content += scanner.Text()
+			c.stdout.mu.Unlock()
+		}
+		if err := scanner.Err(); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	go func() {
+		scanner := bufio.NewScanner(stderrPipe)
+		for scanner.Scan() {
+			c.stderr.mu.Lock()
+			c.stderr.content += scanner.Text()
+			c.stderr.mu.Unlock()
+		}
+		if err := scanner.Err(); err != nil {
+			log.Fatal(err)
+		}
+	}()
+	c.status = "running"
+}
+
+// Wait waits for the command to exit
+func (c *Cmd) Wait() {
+	c.validateHasStarted()
```

Comment:
Let's see. `validateHasStarted()` checks if status == "initialized". However, after running `Start()` the status will be "running". I am assuming that `Wait()` runs after `Start()`, is that not the case?

Reply:
Yes, `validateHasStarted` will raise an error if `status == "initialized". `Wait()` should be called after `Start()`. `Start()` will advance `status` to `running`, so this will only raise if it's called before `Start()`. I'll clarify the comment and add a test.

Reply:
We'll need to move away from `log.Fatal` before this can be tested.